### DISCUSSION TITLE
How to restrict IP from country

### DISCUSSION THREAD

**Discussion by manhthepixta** [vercel/vercel#5304] [Help] [answered]
I want to do following things:
- Restrict IPs from a country (maybe IP range)
- Add an exception as whitelist IP to allow access

How can I do it? I'm using Vercel and Route53

**Comment by paulogdm** [comment]
EDIT: my answer is now outdated! Please refer to our examples and documentations for more information, such as:

- https://vercel.com/docs/security/vercel-waf/rule-configuration
- https://github.com/vercel/examples/tree/main/edge-functions/geolocation-country-block
- https://vercel.com/guides/geo-ip-headers-geolocation-vercel-functions

Hello,

It is not possible to use Vercel to restrict those IPs. However, Cloudflare can support that functionality if you use them as a proxy. Check https://vercel.com/knowledge/using-cloudflare-with-vercel as well.

**Reply by manhthepixta** [reply]
Thanks

**Reply by manhthepixta** [reply]
@paulogdm We're using Cloudfront. Are there any alternatives?

**Comment by PaulRBerg** [comment, marked as answer]
Hello @paulogdm, are there any updates on this?

Is it possible to add a geofence with the latest version of Vercel?

I saw this article: [IP Geolocation now available for all plans](https://vercel.com/changelog/ip-geolocation-now-available-for-all-plans), but I think it only applies to Serverless Functions?

**Reply by leerob** [reply]
https://github.com/vercel/examples/tree/main/edge-functions/geolocation-country-block

**Reply by PaulRBerg** [reply]
Thanks @leerob!
